### PR TITLE
refactor(core): add regex for commonly used fileds

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

refactor(core): add regex for commonly used fields
These rules for fields are temporary and will be replaced with proper rules before v1 is launched and this is tracked by LOG-1630.

## Linear Issue Reference

LOG-1569

## Testing

UT

### CODE REVIEW

File: packages/core/src/utils/regex.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
 export const emailRegEx = /^\S+@\S+\.\S+$/;
 export const phoneRegEx = /^[1-9]\d{10}$/;
-export const usernameRegEx = /^[A-Za-z]\w*$/;
+export const usernameRegEx = /^[A-Za-z]\w{1,17}$/;
+export const nameRegEx = /^[A-Za-z一-龥].{1,17}$/;
+export const passwordRegEx = /^.{6,12}$/;
```
[the remark this answers is not in the document]
Comment (gao-sun) [reply packages/core/src/utils/regex.ts:5]: didn't follow this. we don't restrict password except the minimum length requirement (6 chars) for v1.

cc @xiaoyijun 12 max. is too short

File: packages/core/src/utils/regex.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
 export const emailRegEx = /^\S+@\S+\.\S+$/;
 export const phoneRegEx = /^[1-9]\d{10}$/;
-export const usernameRegEx = /^[A-Za-z]\w*$/;
+export const usernameRegEx = /^[A-Za-z]\w{1,17}$/;
+export const nameRegEx = /^[A-Za-z一-龥].{1,17}$/;
```
[the remark this answers is not in the document]
Comment (gao-sun) [reply packages/core/src/utils/regex.ts:4]: didn't get the regex `-龥` here

File: packages/core/src/utils/regex.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
 export const emailRegEx = /^\S+@\S+\.\S+$/;
 export const phoneRegEx = /^[1-9]\d{10}$/;
-export const usernameRegEx = /^[A-Za-z]\w*$/;
+export const usernameRegEx = /^[A-Za-z]\w{1,17}$/;
```
[the remark this answers is not in the document]
Comment (gao-sun) [reply packages/core/src/utils/regex.ts:3]: i would like to restrict in english letters for username to keep it simple

[the remark this answers is not in the document]
Comment (gao-sun) [reply packages/core/src/utils/regex.ts:3]: @xiaoyijun why do we limit to 18 chars max. when we have length 128 in the table? even if considering unicode later at least we can have 64?

[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/core/src/utils/regex.ts:3]: besides, wondering do we have the minimum length restriction? As previously it was set to min(3)

File: packages/core/src/utils/regex.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/core/src/utils/regex.ts:5]: at this point do we need the regex rule for passwords at this point?

File: packages/core/src/utils/regex.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/core/src/utils/regex.ts:3]: we can just set the minimum length to 3 for v1?

**Review by gao-sun — APPROVED** [review]

**Pull request merged**